### DISCUSSION TITLE
PRs for individual package major version bumps target another baseBranch

### DISCUSSION THREAD

**Discussion by dmt0** [renovatebot/renovate#9291] [Request Help] [answered]
I have a monorepo with a bunch of apps. I want some of the apps to target another branch. I have rules like this:

```
  "baseBranches": ["master", "another-branch"],
  "packageRules": [
    {
      "groupName": "Webapp minor deps",
      "matchPaths": ["path/to/webapp/package.json"],
      "matchUpdateTypes": ["minor", "patch"],
      "matchDepTypes": ["dependencies"],
      "matchBaseBranches": ["another-branch"],
    },
   ...
   ]
```

Which work well for minor/patch version. But Renovate still creates individual major version bump PRs (which is what I want), but all targeting `master` (which is **not** what I want). How would one go about fixing that? If I make a rule for `major` types, it'll just group all majors into one PR, which is not what I want.

**Comment by rarkins** [comment, marked as answer]
> Which work well for minor/patch version. 

As expected, because the rule targets only minor and patch

> But Renovate still creates individual major version bump PRs (which is what I want), 

Which is default behavior

> but all targeting `master` (which is **not** what I want). 

This part doesn't make sense. By default you will get individual major version bump PRs *separate* for each base branch.

> How would one go about fixing that? If I make a rule for `major` types, it'll just group all majors into one PR, which is not what I want.

There's no rule needed for that, and I suspect there's some other confusion, such as rate limiting. Look in your logs for "packageFiles with updates" and you'll get a JSON tree of all the detected dependencies and updates for each base branch. You can see which updates are to be expected. e.g. pick out one particular one you think is missing and see if an update has been identified in the first place.

**Reply by dmt0** [reply]
What I'm really trying to do is stop all PRs for `major` bumps targeting `master` and only have PRs for `major` bumps targeting `another-branch` - but only for some of my apps. Basically to limit the noise.

**Reply by rarkins** [reply]
```
{
  "matchBaseBranches": ["master"],
  "matchUpdateTypes": ["major"],
  "enabled": false
}
```

Just make sure that you don't have any rule after that which re-enables them

**Reply by dmt0** [reply]
Bingo. Thanks!
